### PR TITLE
#51 update kafka base image to bitnami

### PR DESCRIPTION
**Pull request by ewilkins-csi**
Updates the base image for the aissemble-kafka image from wurstmeister kafka to bitnami/kafka:3.5.  The bitnami image was selected because our helm chart, aissemble-kafka-chart, is based on the bitnami kafka chart. The version 3.5 was chosen over the newer 3.7 version image because there was a major refactor to bitnami's chart in chart version 24.0 that would have required a significant refactor and testing to our chart to upgrade beyond. Instead I updated to the last compatible version of the chart, 23.0.7, which was based on the 3.5 image.

### CODE REVIEW

File: build-parent/pom.xml [hunk 1]
```diff
@@ -32,7 +32,6 @@
         <version.python>3.11.4</version.python>
         <version.help.plugin>3.2.0</version.help.plugin>
         <version.krausening>19</version.krausening>
-        <version.license.plugin>2.4.0</version.license.plugin>
```
Comment (ewilkins-csi) [inline build-parent/pom.xml:35]: I: moved to `aissemble-parent`

File: extensions/extensions-helm/aissemble-airflow-chart/pom.xml [hunk 2]
```diff
@@ -15,31 +15,9 @@
 
     <build>
         <plugins>
-            <!-- The airflow subchart defaults to setting the image tag to the airflow version -->
-            <!-- We use a template here to ease release maintenance.  -->
             <plugin>
                 <groupId>com.google.code.maven-replacer-plugin</groupId>
                 <artifactId>replacer</artifactId>
-                <version>1.5.3</version>
```
Comment (ewilkins-csi) [inline extensions/extensions-helm/aissemble-airflow-chart/pom.xml:23]: I: Moved configuration up to the `extensions-helm` POM so we just have to include the plugin to do the version replacement.

File: extensions/extensions-helm/aissemble-kafka-chart/pom.xml [hunk 3]
```diff
@@ -15,6 +15,10 @@
 
     <build>
         <plugins>
+            <plugin>
```
Comment (ewilkins-csi) [inline extensions/extensions-helm/aissemble-kafka-chart/pom.xml:18]: I: Now that we are using our own image again, we need to replace the image tag with the correct version at build time so releasing is smoother.

File: foundation/foundation-maven-plugins/artifacts-maven-plugin/pom.xml [hunk 4]
```diff
@@ -98,7 +98,6 @@
         <dependency>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-dependency-plugin</artifactId>
-            <version>${version.maven.dependency.plugin}</version>
```
Comment (ewilkins-csi) [inline foundation/foundation-maven-plugins/artifacts-maven-plugin/pom.xml:101]: I: Moved to `dependencyManagement` of `aissemble-parent`, but I wasn't convinced that was the right move. Happy to either move it to `build-parent` or just add back these version tags.

File: pom.xml [hunk 5]
```diff
@@ -6,7 +6,7 @@
     <parent>
         <groupId>com.boozallen.aissemble</groupId>
         <artifactId>aissemble-parent</artifactId>
-        <version>2</version>
+        <version>3-SNAPSHOT</version>
```
Comment (ewilkins-csi) [inline pom.xml:9]: I: Can be updated when `aissemble-parent` is released. (See related PR boozallen/aissemble-parent#3)

File: foundation/foundation-maven-plugins/artifacts-maven-plugin/pom.xml (hunk 4 above)
Comment (cwoods-cpointe) [reply foundation/foundation-maven-plugins/artifacts-maven-plugin/pom.xml:101]: I think its fine for now. If a different version is needed in the future then we can override it then

**Review by cwoods-cpointe — APPROVED** [review]

**Review by cpointe-ibllanos — APPROVED** [review]
LGTM

**Review by JeffreyRoss — APPROVED** [review]

**Pull request merged**